### DISCUSSION TITLE
Disappointing range / low RSSI

### DISCUSSION THREAD

**Discussion by jrhbcn** [PricelessToolkit/MailBoxGuard#24] [Q&A] [answered]
Hi,

I have finally finished building my mail sensor. Soldered all component, flashed the firmware and everything was ready to go. Congrats to @PricelessToolkit for all the effort and excellent documentation. Many thanks!

However, I am getting really low and inconsistent radio ranges.  RSSI is just quite low, even when sensor/receiver have line of sight. 

For instance, when both sensor and receiver are next to each other (0m) I already get -75dBm of RSSI. Moving them to 1-2m (no obstacles) drops to -100dBm and just in the other room (3m with a thin wall in between) it already goes to -115dBm. Anything much further than that with a couple of walls stops the communication completely. 

![PXL_20231023_072658979](https://github.com/PricelessToolkit/MailBoxGuard/assets/22198270/4fac93f1-deff-4896-8ade-8be36e193609)

Are these normal values? What RSSI do you get when sensor and receiver are next to each other? My only thoughts are that the sensor antenna is not very good, do you recommend an antenna I can buy to better test this?

I was expecting to use the sensor to read the mailbox in my flat (entrance to 4th floor). Are my expectations of using it in a 4th floor completely unrealistic? :( 

Again, thanks a lot for this project!

**Comment by jrhbcn** [comment]
Just thought about something now.  I see that the RA-01 module specifies 410-525Mhz while I am using the 868Mhz band (as the LILYGO receiver states). Is that the correct component to use in the 868Mhz band? Maybe I do need another version of the LORA chip for the sensor? I did not think about this and just thought all RA-01 components adapted to different frequencies in software.

**Comment by jrhbcn** [comment]
Edit 3: I am pretty sure now my problem is that module ... arggg I should have gone to the RA-01H version. I'll try to desolder and buy the other module, but my previous desolder was awful and need to re-do everything. :'''(

**Reply by PricelessToolkit** [reply]
yep, you need RA-01H, you can desolder with a hot plate or with hot air easily! Just take your time!.

PS: why you don't cut the charger part?

**Reply by PricelessToolkit** [reply]
and one more thing you need a different spring antenna too, for 868 MHz antenna is 2 times shorter

**Reply by jrhbcn** [reply]
Yes, I am sooo dumb XD  RA-01H module it is (just bought it in aliexpress so several weeks to wait). I will indeed cut the charger part but I was still unsure if I wanted to reuse the battery you see on the photo or to buy some new ones like you recommended in your video :) 

I have also ordered some 868Mhz antennas with it. Thanks.

So do you think entrance to 4th floor is realistic (with proper modules for sure!)?

PS: Also, I have created some folders to compile both the sensor and ttygo boards using platformio and visual studio code in a fork. If you're interested, I can prepare a small PR.

**Reply by jrhbcn** [reply]
>  you can desolder with a hot plate or with hot air easily!

I only have a hot plate (not hot air) and I did de-solder other components last time, I am afraid of the attiny1616 small chip close to the LORA one... will see!!

**Reply by PricelessToolkit** [reply]
one time i ordered 50x RA-01S insteed of RA-01, so i am dumest! :), 4 floor is not a problem, I am 99.9% sure. When desoldering don't forget to add flux and don't panic, just wait and pock with tweezers until you see it moving, then remove it.

**Reply by jrhbcn** [reply, marked as answer]
OK, I just wanted to say that finally the new chip arrived and everything works like a charm. I get between -85dBm and -90dBm with 4 floors between the sensor and the gateway. 

Many thanks for publishing this nice project.

**Reply by PricelessToolkit** [reply]
I'm happy everything went well! Maybe you'd like to assemble my latest project too? You can find it here: https://github.com/PricelessToolkit/ESPBell-MAX  I'll make the PCB files available when I release the Video.
